File: Linked-List-Length/Linked-List-Length.cpp
#include <iostream>
using namespace std;

struct Node {
    int data;
    Node* next;

    // Constructor to initialize a node with data
    Node(int value) : data(value), next(nullptr) {}
};

// Function to count the number of nodes in the linked list
int countListSize(Node* head) {
    int size = 0;
    Node* temp = head;

    while (temp != nullptr) {
        size++;  // Increment size for each node
        temp = temp->next;  // Move to the next node
    }

    return size;
}

int main() {
    // Create a linked list: 1 -> 2 -> 3
    Node* head = new Node(1);
    head->next = new Node(2);
    head->next->next = new Node(3);
    head->next->next->next = new Node(5);

    // Count the number of nodes in the list
    int size = countListSize(head);
    cout << "Number of nodes in the list: " << size << endl;

    // Clean up memory
    Node* temp;
    while (head != nullptr) {
        temp = head;  // Store the current head
        head = head->next;  // Move the head to the next node
        delete temp;  // Delete the old head
    }

    return 0;
}

File: Linked-List-Length/LinkedListLength.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

# Function to count the number of nodes in the linked list
def count_list_size(head):
    size = 0
    temp = head
    while temp is not None:
        size += 1  # Increment size for each node
        temp = temp.next  # Move to the next node
    return size

if __name__ == "__main__":
    # Create a linked list: 1 -> 2 -> 3 -> 5
    head = Node(1)
    head.next = Node(2)
    head.next.next = Node(3)
    head.next.next.next = Node(5)

    # Count the number of nodes in the list
    size = count_list_size(head)
    print("Number of nodes in the list:", size)

    # Clean up memory (not necessary in Python due to automatic garbage collection)
    head = None  # Just set head to None to break the links, and Python's GC will take care of the rest
